actors/miner/tests/prove_commit.rs

```diff
@@ -340,3 +354,343 @@ fn prove_sectors_from_batch_pre_commit() {
+    }
+}
+
+#[test]
+fn invalid_proof_rejected() {
+    let h = ActorHarness::new(PERIOD_OFFSET);
+    let mut rt = h.new_runtime();
+    rt.balance.replace(TokenAmount::from(BIG_BALANCE));
+
+    let precommit_epoch = PERIOD_OFFSET + 1;
+    rt.set_epoch(precommit_epoch);
+
+    h.construct_and_verify(&mut rt);
+    let deadline = h.deadline(&rt);
+
+    // Make a good commitment for the proof to target.
+    let sector_no = 100;
+    let params = h.make_pre_commit_params(
+        sector_no,
+        precommit_epoch - 1,
+        deadline.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period,
+        vec![1],
+    );
+    let precommit = h.pre_commit_sector(&mut rt, params, PreCommitConfig::empty(), true);
+
+    // Sector pre-commitment missing.
+    rt.set_epoch(precommit_epoch + rt.policy.pre_commit_challenge_delay + 1);
+    expect_abort(
+        ExitCode::USR_NOT_FOUND,
+        h.prove_commit_sector_and_confirm(
+            &mut rt,
+            &precommit,
+            h.make_prove_commit_params(sector_no + 1),
+            ProveCommitConfig::empty(),
+        ),
+    );
+    rt.reset();
+
+    // Too late.
+    rt.set_epoch(
+        precommit_epoch + max_prove_commit_duration(&rt.policy, precommit.info.seal_proof) + 1,
+    );
+    expect_abort(
+        ExitCode::USR_NOT_FOUND,
+        h.prove_commit_sector_and_confirm(
+            &mut rt,
+            &precommit,
+            h.make_prove_commit_params(sector_no + 1),
+            ProveCommitConfig::empty(),
+        ),
+    );
+    rt.reset();
+
+    // Too early.
+    rt.set_epoch(precommit_epoch + rt.policy.pre_commit_challenge_delay - 1);
+    expect_abort(
+        ExitCode::USR_NOT_FOUND,
+        h.prove_commit_sector_and_confirm(
+            &mut rt,
+            &precommit,
+            h.make_prove_commit_params(sector_no + 1),
+            ProveCommitConfig::empty(),
+        ),
+    );
+    rt.reset();
+
+    // Set the right epoch for all following tests
+    rt.set_epoch(precommit_epoch + rt.policy.pre_commit_challenge_delay + 1);
+
+    // Invalid deals (market ActivateDeals aborts)
+    let verify_deals_exit =
+        HashMap::from([(precommit.info.sector_number, ExitCode::USR_ILLEGAL_ARGUMENT)]);
+    expect_abort(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        h.prove_commit_sector_and_confirm(
+            &mut rt,
+            &precommit,
+            h.make_prove_commit_params(sector_no),
+            ProveCommitConfig { verify_deals_exit },
+        ),
+    );
+    rt.reset();
+
+    rt.balance.replace(TokenAmount::from(1_000) * 1e18 as u64);
+
+    let prove_commit = h.make_prove_commit_params(sector_no);
+    h.prove_commit_sector_and_confirm(
+        &mut rt,
+        &precommit,
+        prove_commit,
+        ProveCommitConfig::empty(),
+    )
+    .unwrap();
+    let st = h.get_state(&rt);
+
+    // Verify new sectors
+    // TODO minerstate
```

Comment:
This is missing

Reply:
It is missing in the specs-actors code as well. I copied the comments verbatim. Fixing this TODO is beyond the scope of this PR.